src/App.js

```diff
@@ -14,11 +15,84 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    isLoad: false,
+    goods: [...goodsFromServer],
+    goodsSelected: [...goodsFromServer],
+    minLength: '1',
+  };
+
+  checkButton = () => {
+    this.setState({
+      isLoad: true,
+    });
+  };
+
+  handleReverseList = () => {
+    this.setState(prevState => ({
+      goods: [...prevState.goods].reverse(),
+    }));
+  };
+
+  handleSortAlphabet = () => {
+    this.setState(prevState => ({
+      goods: [...prevState.goods].sort((a, b) => a.localeCompare(b)),
+    }));
+  };
+
+  handleResetList = () => {
+    this.setState(prevState => ({
+      minLength: '1',
+      goods: [...prevState.goodsSelected],
+    }));
+  };
+
+  handleSortByLength = () => {
+    this.setState(prevState => ({
+      goods: [...prevState.goods].sort((a, b) => a.length - b.length),
+    }));
+  };
+
+  handleSelectList = ({ target }) => {
+    this.setState(prevState => ({
+      minLength: target.value,
+      goods: [...prevState.goodsSelected]
+        .filter(good => good.length >= target.value),
+    }));
+  };
+
+  render() {
+    const { goods, minLength, isLoad } = this.state;
+
+    return (
+      <div className="App">
```

Comment:
классы с маленькой буквы
```suggestion
      <div className="app">
```